airflow-core/src/airflow/api_fastapi/core_api/routes/ui/dependencies.py

```diff
@@ -41,8 +46,24 @@
     ),
     dependencies=[Depends(requires_access_dag("GET", DagAccessEntity.DEPENDENCIES))],
 )
-def get_dependencies(session: SessionDep, node_id: str | None = None) -> BaseGraphResponse:
+def get_dependencies(
+    session: SessionDep,
+    node_id: str | None = None,
+    dependency_type: Literal["scheduling", "data"] = "scheduling",
+) -> BaseGraphResponse:
     """Dependencies graph."""
+    if dependency_type == "data":
+        if node_id is None or not node_id.startswith("asset:"):
+            raise HTTPException(400, "Data dependencies require an asset node_id (e.g., 'asset:123')")
+
+        try:
+            asset_id = int(node_id.replace("asset:", ""))
+        except ValueError:
+            raise HTTPException(400, f"Invalid asset node_id: {node_id}")
```

Comment:
Just to align the `HTTPException` usage with the other routers.

```suggestion
            raise HTTPException(status.HTTP_400_BAD_REQUEST, "Data dependencies require an asset node_id (e.g., 'asset:123')")

        try:
            asset_id = int(node_id.replace("asset:", ""))
        except ValueError:
            raise HTTPException(status.HTTP_400_BAD_REQUEST, f"Invalid asset node_id: {node_id}")
```